jupyterhub/tests/selenium/test_browser.py

```diff
@@ -209,36 +225,686 @@ async def test_open_url_login(
     try:
         error = browser.find_element(*LoginPageLocators.ERROR_INVALID_CREDANTIALS)
         await webdriver_wait(browser, EC.visibility_of(error))
     except NoSuchElementException:
         error = None
 
-    # verify error message and url still eguals to the login page
+    # verify error message and url
     assert LoginPageLocators.ERROR_MESSAGES_LOGIN == error.text
     assert 'hub/login' in browser.current_url
 
 
-# HOME PAGE
-async def open_home_page(app, browser, user="test_user", pass_w="test_user"):
+# MENU BAR
+
+
+@pytest.mark.parametrize(
+    "value, user, pass_w",
+    [
+        # the home page: verify if links work on the top bar
+        ("/hub/home", "test_user", "test_user"),
+        # the token page: verify if links work on the top bar
+        ("/hub/token", "test_user", "test_user"),
+        # the spawn-pending page: verify if links work on the top bar
+        ("hub/spawn", "test_user", "test_user"),
+        # "hub/not" = any url that is not existed: verify if links work on the top bar
+        ("hub/not", "test_user", "test_user"),
+        # the login page: verify if links work on the top bar
+        ("", None, None),
+    ],
+)
+async def test_menu_bar(app, browser, slow_spawn, value, user, pass_w):
+    next_page = ujoin(url_escape(app.base_url) + value)
+    await open_url(app, browser, param="/login?next=" + next_page)
+    if value:
+        await login(browser, user, pass_w)
+    await webdriver_wait(
+        browser, EC.presence_of_all_elements_located(BarLocators.LINK_HOME_BAR)
+    )
+    logo = browser.find_element(*BarLocators.LOGO)
+    # verify the title on the logo
+    assert logo.get_attribute('title') == "Home"
+
+    links_bar = browser.find_elements(*BarLocators.LINK_HOME_BAR)
+    if not value:
+        assert len(links_bar) == 1
+    elif "hub/not" in value:
+        assert len(links_bar) == 3
+    else:
+        assert len(links_bar) == 4
+        user_name = browser.find_element(*BarLocators.USER_NAME)
+        assert user_name.text == user
+
+    for index in range(len(links_bar)):
+
+        await webdriver_wait(
+            browser, EC.presence_of_all_elements_located(BarLocators.LINK_HOME_BAR)
+        )
+        links_bar = browser.find_elements(*BarLocators.LINK_HOME_BAR)
+
+        links_bar_url = links_bar[index].get_attribute('href')
+        links_bar_title = links_bar[index].text
+        links_bar[index].click()
+        await asyncio.sleep(0.5)
+        # verify that links on the topbar work, checking the titles of links
+        if index == 0:
+            assert links_bar_url.endswith("/hub/")
+            if not value:
+                assert (
+                    f"hub/login?next={url_escape(app.base_url)}" in browser.current_url
+                )
+            else:
+                u = app.users[orm.User.find(app.db, user)]
+                while not u.spawner.ready:
+                    await asyncio.sleep(1.5)
```

Comment:
Let's shorten the interval:

```suggestion
                    await asyncio.sleep(0.1)
```